src/VisualStudio/Core/Def/Implementation/Progression/GraphProvider.cs

```diff
@@ -136,11 +137,15 @@ private void EnsureInitialized()
 
                 if (searchParameters != null)
                 {
-                    // WARNING: searchParameters.SearchQuery returns an IVsSearchQuery object, which
-                    // is a COM type. Therefore, it's probably best to grab the values we want now
-                    // rather than get surprised by COM marshalling later.
-                    graphQueries.Add(new SearchGraphQuery(
-                        searchParameters.SearchQuery.SearchString, threadingContext, asyncListener));
+                    // WARNING: searchParameters.SearchQuery returns an IVsSearchQuery object, which is a COM type.
+                    // Therefore, it's probably best to grab the values we want now rather than get surprised by COM
+                    // marshalling later.
+                    //
+                    // Create two queries.  One to find results in normal docs, and one to find results in generated
+                    // docs.  That way if the generated docs take a long time we can still report the regular doc
+                    // results immediately.
+                    graphQueries.Add(new SearchGraphQuery(searchParameters.SearchQuery.SearchString, NavigateToSearchScope.RegularDocuments, threadingContext, asyncListener));
```

Comment:
similar question as above, the code was already searching non-source generated first and since we're streaming results back wouldn't they have been reported first?  why do we need to explicitly separate the search?

Reply:
> similar question as above, the code was already searching non-source generated first and since we're streaming results back wouldn't they have been reported first? why do we need to explicitly separate the search?

Ah, i see the question.  Progression is *not* streaming.  We have to grab the produced-sub-graph (what we make) and apply it at once to the full graph.  So this allows us to break that into two chunks.  The produced-sub-graph for teh normal docs, and the produced-sub-graph for the generated docs.  we can apply each to the larger graph when each finishes.